PowerValidatedSolutions.psm1

```diff
@@ -10386,6 +10386,151 @@ Function Undo-vRLIAlert {
+
+        .EXAMPLE
+        Add-vRLILogForwarder -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -name lax-vrli01 -fqdn lax-vrli01.lax.rainpole.io -protocol CFAPI -port 9543 -acceptCert true -sslEnabled true -testConnection true
+        This example adds a log forwarder destination to vRealize Log Insight
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$name,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+        [Parameter (Mandatory = $true)] [ValidateSet("CFAPI", "SYSLOG", "RAW")] [String]$protocol,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$port,
+        [Parameter (Mandatory = $false)] [ValidateSet("TCP", "UDP")] [String]$transport,
+        [Parameter (Mandatory = $true)] [ValidateSet('true', 'false')] [String]$acceptCert,
+        [Parameter (Mandatory = $true)] [ValidateSet('true', 'false')] [String]$sslEnabled,
+        [Parameter (Mandatory = $true)] [ValidateSet('true', 'false')] [String]$testConnection
+    )
+
+    Try {
+        if (Test-VCFConnection -server $server) {
+            if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+                if (($vcfVrliDetails = Get-vRLIServerDetail -fqdn $server -username $user -password $pass)) {
+                    if (Test-vRLIConnection -server $vcfVrliDetails.fqdn) {
+                        if (Test-vRLIAuthentication -server $vcfVrliDetails.fqdn -user $vcfVrliDetails.adminUser -pass $vcfVrliDetails.adminPass) {
+                            if (! (Get-vRlILogForwarder | Where-Object {$_.host -eq $fqdn})) {
+                                if ($protocol -eq 'SYSLOG' -and (-not $PsBoundParameters.ContainsKey('transport'))) {
+                                    Throw 'You must enter a transport for SYSLOG.'
+                                } elseif ($protocol -eq 'SYSLOG' -and ($PsBoundParameters.ContainsKey('transport'))) {
+                                    Set-vRLILogForwarder -name $name -server $fqdn -protocol $protocol -port $port -transport $transport -acceptCert $acceptCert -sslEnabled $sslEnabled -testConnection $testConnection | Out-Null
+                                    if (Get-vRlILogForwarder | Where-Object {$_.host -eq $fqdn}) {
+                                        Write-Output "Adding log forwarder ($fqdn) to vRealize Log Insight ($($vcfVrliDetails.fqdn)): SUCCESSFUL"
+                                    } else {
+                                        Write-Warning "Adding log forwarder ($fqdn) to vRealize Log Insight ($($vcfVrliDetails.fqdn)): FAILED"
```

Comment:
FAILED should be POST_VALIDATION_FAILED

Reply:
Addressed in commit 0d5d50626f7349f08f82e03938ac9df41c359695.